src/utils/HealthLogger.js

```diff
@@ -200,6 +202,39 @@ define(function (require, exports, module) {
         setHealthDataLog("searchDetails", searchDetails);
     }
 
+    /**
+     * Send Analytics Data
+     * @param {string} eventCategory The kind of Event Category that
+     * needs to be logged- should be a js var compatible string
+     * @param {string} eventSubCategory The kind of Event Sub Category that
+     * needs to be logged- should be a js var compatible string
+     * @param {string} eventType The kind of Event Type that needs to be logged- should be a js var compatible string
+     * @param {string} eventSubType The kind of Event Sub Type that
+     * needs to be logged- should be a js var compatible string
+     */
+    function sendAnalyticsData(eventName, eventCategory, eventSubCategory, eventType, eventSubType) {
+        var isEventDataAlreadySent = PreferencesManager.getViewState(eventName);
+        //Parameters containing event information
+        var eventParams = {};
+        if(!isEventDataAlreadySent) {
+            eventParams =  {
+                eventName: eventName || "",
```

Comment:
Maybe we should not send data if there is no event name and category? I think we can check for these params. And for rest param can be optional. i.e.

```
if (eventName && eventCategory) {
            eventParams =  {
                eventName: eventName,
                eventCategory: eventCategory,
                eventSubCategory: eventSubCategory || "",
                eventType: eventType || "",
                eventSubType: eventSubType || ""
            };
            notifyHealthManagerToSendData(eventParams);
}
```